### ISSUE TITLE
Support user provided services

### ISSUE THREAD

**Issue by geofffranks** [jhunt/cf-plugin-deploy#5]
We should support defining/updating user provided services via cf-plugin-deploy.

Possible config format (underneath the `spaces` section):
```
organizations:
  my-org:
    spaces:
      my-space:
        user-provided-services:
        - name: my-user-provided-service
           credentials:
             key: value
             other: value
        - name: my-user-provided-route-service
           route_service_url: https://example.com
        - name: my-user-provided-syslog-drain
           syslog_drain_url: syslog://example.com
```

These services should also be treated like regular services in `cf-deploy` so that they can be bound to applications when they're pushed via cf-deploy.

**Comment by geofffranks** [comment]
Seems like we should support any combination of credentials, route_service_url, and syslog_drain_url.  At least one should be present, but all three could be provided on a given service

**Comment by brijeshrakholia** [comment]
Seems like there are three ways of creating user provided services for credentials option: 

1. A simple string which needs user inputs (this can pause the automation)
2. You provide a JSON string 
3. You provide a /path/to/credentials.json

Which of these three do we want to support? @norman-abramovitz and my thoughts: We should only support 2 & 3. What do you think?

**Comment by geofffranks** [comment]
I was thinking we would just take the contents of the `credentials` object in the YAML file, and JSON-ify it.  Adding file support seems redundant, since you could embed the same datastructure inside the YAML file, so the indirection is less useful.

**Comment by jhunt** [comment]
Fixed in #7
